### PR TITLE
Update version React-Native-Onyx 1.0.39

### PR DESCRIPTION
**Pull request by tienifr**
### Details

It comes from [this PR](https://github.com/Expensify/App/pull/15556)

### Fixed Issues

$ https://github.com/Expensify/App/issues/15321
$ https://github.com/Expensify/App/issues/15321#issuecomment-1441602988

### Tests

1. Open the app in the browser
2. Open another tab in the same browser
3. Log out from one of the tabs 
4. Verify that in other tab the app is not crashed and logged out to the login page.

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Open the app in the browser
2. Open another tab in the same browser
3. Disable the internet
3. Log out from one of the tabs 
4. Verify that in other tab the app is not crashed and logged out to the login page.

### QA Steps

1. Open the app in the browser
2. Open another tab in the same browser
3. Log out from one of the tabs 
4. Verify that in other tab the app is not crashed and logged out to the login page.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/113963320/232407746-f761242b-eae6-40ec-9c7f-1ed599bde518.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/113963320/232412867-13cfc942-33ff-42bc-bdc7-7a640d632928.mp4

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/113963320/232417228-48cb8c7b-4287-4dc9-b33c-52dc71f61303.mp4

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/113963320/232407075-4d5f3c4e-2f5c-4285-b984-a85435013835.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/113963320/232407120-ec41578e-e8b0-4b70-b42c-819e9e020aa1.mov

</details>

<details>
<summary>Android</summary>

[15321 android.webm](https://user-images.githubusercontent.com/113963320/232407205-7430c8ef-6741-4e74-990e-0421bf571365.webm)

</details>

### CODE REVIEW

**Comment by tienifr** [comment]
Hi @sobitneupane @roryabraham This PR update the React-Native-ONYX version to 1.0.39. Please help to review this PR first

File: package.json
Comment (sobitneupane) [inline package.json:110]: Why is this change required here?

Comment (tienifr) [reply package.json:110]: In the react-Native-Onyx, we just updated "react-native-quick-sqlite": "^8.0.0-beta.2" and this package is peerDependencies so we need update it in the App repo

Comment (tienifr) [reply package.json:110]: Yes, it is [this PR](https://github.com/Expensify/react-native-onyx/pull/235)

**Review by sobitneupane — COMMENTED** [review]
Yet to test on android. Getting some errors while building android app.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/25876548/228179765-3d707bb7-1c91-4b94-89bf-76cc6cc459bd.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/25876548/228179922-7feae205-3590-4fac-bcf4-ef02edddff0b.mov



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://user-images.githubusercontent.com/25876548/228179865-730c3035-ab46-48b0-bd25-4056abf0917b.mov


</details>

<details>
<summary>Desktop</summary>


https://user-images.githubusercontent.com/25876548/228179808-148fe918-4860-4119-9afc-70e3667850fa.mov


</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/25876548/228179889-4d6e6c17-3c18-46fa-8e26-01b3c147ce05.mov



</details>

<details>
<summary>Android</summary>


</details>

**Comment by tienifr** [comment]
@sobitneupane 
This bug causes by upgrading the version of "react-native-quick-sqlite": "^8.0.0-beta.2" that is [merged](https://github.com/Expensify/react-native-onyx/pull/235) in React-Native-ONYX but it has not been updated in the app repo
So I think we need to wait for a PR update React-Native-ONYX version to 1.0.38 from [this issue](https://github.com/Expensify/App/issues/15124)

**Comment by priyeshshah11** [comment]
> @sobitneupane This bug causes by upgrading the version of "react-native-quick-sqlite": "^8.0.0-beta.2" that is [merged](https://github.com/Expensify/react-native-onyx/pull/235) in React-Native-ONYX but it has not been updated in the app repo So I think we need to wait for a PR update React-Native-ONYX version to 1.0.38 from [this issue](https://github.com/Expensify/App/issues/15124)
> 
> cc @priyeshshah11 @mountiny

Commented here https://github.com/Expensify/App/issues/15124#issuecomment-1491297990

**Review by roryabraham — DISMISSED** [review]
@tienifr LGTM, but looks like we've got conflicts

**Comment by mountiny** [comment]
I think this needs to be on hold for the RN update @roryabraham

**Review by mountiny — COMMENTED** [review]
@tienifr The RN update has been merged, would you be able to resolve the merge conflicts? I think we should wait one release cycle before merging so this is not merged together with the RN update in case there are some regressions from either.

**Comment by mountiny** [comment]
@tienifr bumping this one, I think we are good to go now

**Comment by mountiny** [comment]
@sobitneupane would you be able to retest please 🙇

**Review by mountiny — APPROVED** [review]
LGTM, lets test this and get the checklist done and we can :shipit:

**Comment by sobitneupane** [comment]
@tienifr Can you please update the Screenshots.

**Review by sobitneupane — COMMENTED** [review]
### Screenshots/Videos
<details>
<summary>Web</summary>


https://user-images.githubusercontent.com/25876548/232391518-01558436-a022-4810-bba9-c0720b300d80.mov


</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/25876548/232399479-2f4187bd-80fb-4ac1-9f82-47bf87e33de4.mov



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://user-images.githubusercontent.com/25876548/232400910-22fae077-551d-4cdb-ade0-9dc6b416abab.mov


</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/25876548/232401517-14e0c7a3-f1f8-4574-ae61-7d7a8c63fa7a.mov

</details>

<details>
<summary>iOS</summary>


https://user-images.githubusercontent.com/25876548/232401526-cb00f5cd-5c0d-465a-bb0a-be6230216a44.mov

</details>

<details>
<summary>Android</summary>


https://user-images.githubusercontent.com/25876548/232392045-874e4e50-684b-4355-b3aa-e253fe26d89b.mov


</details>

**Comment by sobitneupane** [comment]
@tienifr Bump on https://github.com/Expensify/App/pull/16531#issuecomment-1508155012

**Comment by tienifr** [comment]
@sobitneupane I've updated the screenshot please help take a look. Thanks

**Review by mountiny — DISMISSED** [review]
I think this is looking good based on the @sobitneupane testing, can you also please finish the checklist now that @tienifr updated the screenshots?

File: tests/unit/NetworkTest.js
Comment (roryabraham) [inline tests/unit/NetworkTest.js:35]: Why is this change here?

Comment (mountiny) [reply tests/unit/NetworkTest.js:35]: Oh I thought thats part of it cc @tienifr

Comment (tienifr) [reply tests/unit/NetworkTest.js:35]: In https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/src/libs/Network/NetworkStore.js#L28 we're using resolveIsReadyPromise to run resolve when this page is loaded and we call `resolveIsReadyPromise` to execute resolve function in `checkRequiredData`. 

 https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/tests/unit/NetworkTest.js#L44 
we create a new promise and a new resolve function => we have to call `checkRequiredData` again in beforeEach.

- Why does the previous version works?
before each test run, we call `Onyx.clear()` that update subscriber even the data doesn't change => checkRequiredData is called
https://github.com/Expensify/App/blob/88644166f51c9f7ea8211df0b9afee1ca4784700/src/libs/Network/NetworkStore.js#L57

Comment (tienifr) [reply tests/unit/NetworkTest.js:35]: @mountiny @roryabraham What do you think about my explanation above?

Comment (tienifr) [reply tests/unit/NetworkTest.js:35]: @mountiny @roryabraham Any updates?

**Comment by sobitneupane** [comment]
@tienifr Can you please update https://github.com/Expensify/App/pull/15556 and make it ready for review. I would like to test it before this PR gets merged?

File: tests/unit/NetworkTest.js
Comment (roryabraham) [reply tests/unit/NetworkTest.js:35]: Thanks for explaining

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Comment by tienifr** [comment]
@roryabraham I've changed `localforage-removeitems` order in package.json. 
@sobitneupane I don't actually understand your mean. I think the PR: https://github.com/Expensify/App/pull/15556 is ready for testing (includes all the changes of this PR). Please correct me if i'm wrong. Thanks

**Review by roryabraham — APPROVED** [review]

**Comment by roryabraham** [comment]
LGTM, thanks @tienifr 🙇🏼

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by mountiny — APPROVED** [review]
Nice I think we are good to go, thanks!

**Pull request merged**
